gix-archive/src/lib.rs

```diff
@@ -1,2 +1,62 @@
 #![deny(rust_2018_idioms)]
 #![forbid(unsafe_code)]
+
+mod archive {
+
+    use gix_hash::oid;
+    use gix_object::TreeRefIter;
+    use std::io::Write;
+    use std::path::PathBuf;
+    use std::vec::IntoIter;
+
+    #[derive(Debug, thiserror::Error)]
+    #[allow(missing_docs)]
+    #[error("{msg}")]
+    pub struct FailedToArchiveError {
+        msg: String,
+    }
+
+    /// Available formats
+    #[derive(PartialEq)]
+    pub enum Format {
+        Tar,
+        Zip,
+        TarGz,
+        Tgz,
+        Custom(FromConfig),
+    }
+
+    #[derive(PartialEq)]
+    pub struct FromConfig {
+        name: String,
+        command: String,
+    }
+
+    /// Archive options, such as format, and later prefix, work-tree, etc
+    pub struct Options {
```

Comment:
> After taking another look at possible options, it looks like there will be nothing else. Thus `Options` can be removed and `Format` is all there is left. Everything related to attributes and filters will be so essential that it can't be defaulted or be passed as options.

Other options are --worktree-attributes, --mtime, --prefix